src/components/Context.js

```diff
@@ -235,30 +273,24 @@ class ProductProvider extends Component {
     );
   };
 
-  countDiscount = () => {
-    const voucher = this.state.promotionItems;
-    const actualCartSubTotal = parseInt(this.state.cartSubTotal, 10);
-    const cartTax = this.state.cartTax;
-    let voucherDiscount = 0;
-    let cartDiscount = 0;
+  checkDiscount = () => {
```

Comment:
the logic of this function should also be incorporated into that central `calculateCart()` reducer function